clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -1425,8 +1425,37 @@ private ControllerClient getControllerClient(
   }
 
   private Optional<ByteBuffer> getCompressionDictionary() throws VeniceException {
-    ByteBuffer compressionDictionary = null;
+    if (!pushJobSetting.isZstdDictCreationRequired) {
+      return Optional.empty();
+    }
+
+    ByteBuffer compressionDictionary;
+    try {
+      compressionDictionary = fetchOrBuildCompressionDictionary();
+    } catch (Exception e) {
+      LOGGER.warn("Failed to fetch or build compression dictionary", e);
+      compressionDictionary = null;
+    }
+
+    if (compressionDictionary != null && compressionDictionary.remaining() > 0) {
+      pushJobSetting.isZstdDictCreationSuccess = true;
+      return Optional.of(compressionDictionary);
+    }
+
+    if (pushJobSetting.storeCompressionStrategy != CompressionStrategy.ZSTD_WITH_DICT) {
+      LOGGER.info(
+          "No dictionary fetched. But since the compression strategy is not {}, it is not required",
+          CompressionStrategy.ZSTD_WITH_DICT);
+      pushJobSetting.isZstdDictCreationSuccess = false;
+      return Optional.empty();
+    }
+
+    LOGGER.info("No dictionary fetched. Creating a default dictionary since compression strategy is ZSTD_WITH_DICT");
```

Comment:
In this branch would we get any alert/notification without diving deep into the logs? I'm worried that if `CompressionStrategy.ZSTD_WITH_DICT` is enabled but we failed to fetch due to some random reason wouldn't it impact compression performance without us knowing about it?

Reply:
Yes, it affects the compression ratios. But the alternative for `ZSTD_WITH_DICT` (without a lot of refactoring) is to fail the push job. In case the compression dictionary is bad for the store, the push will fail due to quota exceeding

Reply:
I see, give the push a chance with default dictionary does sound better for now. We might want to document this as a comment or something so we can add more signals if dictionary fetch fails frequently.